case_studies/shallow_water/serial/galewsky_serial.py

```diff
@@ -68,85 +70,138 @@
+        'local_type': 'additive',
+        'precompute_element_tensors': True,
+        'symmetrise_sweep': False
+    },
+    'sub': {
+        'ksp_type': 'preonly',
+        'pc_type': 'lu',
+        'pc_factor_shift_type': 'nonzero',
+    }
+}
+
+mg_params = {
+    'transfer_manager': f'{__name__}.ManifoldTransferManager',
+    'levels': {
+        'ksp_type': 'gmres',
+        'ksp_max_it': 3,
+        'pc_type': 'python',
+        'pc_python_type': 'firedrake.PatchPC',
+        'patch': patch_params
+    },
+    'coarse': {
+        'pc_type': 'python',
+        'pc_python_type': 'firedrake.AssembledPC',
+        'assembled': lu_params
+    },
+}
+
+mg_sparameters = {
+    'mat_type': 'matfree',
+    'pc_type': 'mg',
+    'pc_mg_cycle_type': 'v',
+    'pc_mg_type': 'multiplicative',
+    'mg': mg_params
+}
+
+hybridscpc_sparameters = {
+    'mat_type': 'matfree',
+    'pc_type': 'python',
+    'pc_python_type': f'{__name__}.HybridisedSCPC',
+    'hybridscpc_condensed_field': lu_params,
+    'hybridscpc_condensed_field_snes': linear_snes_params,
+}
+
+aux_sparameters = {
+    'mat_type': 'matfree',
+    'pc_type': 'python',
+    'pc_python_type': 'asQ.AuxiliaryRealBlockPC',
+    'aux': hybridscpc_sparameters,
+    'aux_snes': linear_snes_params
+}
+
+atol = 1e6
 sparameters = {
     'snes': {
         'monitor': None,
         'converged_reason': None,
         'rtol': 1e-12,
-        'atol': 1e-0,
+        'atol': atol,
         'ksp_ew': None,
         'ksp_ew_version': 1,
     },
-    'mat_type': 'matfree',
     'ksp_type': 'fgmres',
     'ksp': {
         'monitor': None,
-        'converged_reason': None,
-        'atol': 1e-5,
+        'converged_rate': None,
+        'atol': atol,
         'rtol': 1e-5,
     },
-    'pc_type': 'mg',
-    'pc_mg_cycle_type': 'w',
-    'pc_mg_type': 'multiplicative',
-    'mg': {
-        'transfer_manager': f'{__name__}.ManifoldTransferManager',
-        'levels': {
-            'ksp_type': 'gmres',
-            'ksp_max_it': 5,
-            'pc_type': 'python',
-            'pc_python_type': 'firedrake.PatchPC',
-            'patch': {
-                'pc_patch_save_operators': True,
-                'pc_patch_partition_of_unity': True,
-                'pc_patch_sub_mat_type': 'seqdense',
-                'pc_patch_construct_dim': 0,
-                'pc_patch_construct_type': 'vanka',
-                'pc_patch_local_type': 'additive',
-                'pc_patch_precompute_element_tensors': True,
-                'pc_patch_symmetrise_sweep': False,
-                'sub_ksp_type': 'preonly',
-                'sub_pc_type': 'lu',
-                'sub_pc_factor_shift_type': 'nonzero',
-            },
-        },
-        'coarse': {
-            'pc_type': 'python',
-            'pc_python_type': 'firedrake.AssembledPC',
-            'assembled_pc_type': 'lu',
-            'assembled_pc_factor_mat_solver_type': 'mumps',
-        },
-    }
 }
+sparameters.update(aux_sparameters)
```

Comment:
can you switch the params with a command argument?

Reply:
de56cf75a84de04a6c27d56c8349485cf4e0a9f0